base_layer/wallet/src/transaction_service/storage/models.rs

```diff
@@ -334,3 +343,50 @@ impl From<WalletTransaction> for CompletedTransaction {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TxCancellationReason {
+    Unknown,            // 0
+    UserCancelled,      // 1
+    Timeout,            // 2
+    DoubleSpend,        // 3
+    Orphan,             // 4
+    TimeLocked,         // 5
+    InvalidTransaction, // 6
+    AbandonedCoinbase,  // 7
+}
```

Comment:
Discussion point for future reference:
- It might be cool to provide a bit more information to the user here, like the value of the time out or identifying which UTXO is implicated in the double spend when the culprit is just one of many dust inputs.